resolver/dns/dns_resolver.go

```diff
@@ -24,10 +24,29 @@
 package dns
 
 import (
+	"time"
+
 	"google.golang.org/grpc/internal/resolver/dns"
 	"google.golang.org/grpc/resolver"
 )
 
+// SetResolvingTimeout sets the maximum duration for DNS resolution requests.
+//
+// This function updates the global `ResolvingTimeout` variable, which specifies
+// how long a DNS resolution attempt will wait before it is canceled.
+//
+// It is recommended to call this function at application startup, before any
+// gRPC calls are made. Modifying this value after initialization is not
+// thread-safe.
+//
+// The default value is 10 seconds. Setting the timeout too low may result
+// in premature timeouts during resolution, while setting it too high may
+// lead to unnecessary delays in service discovery. Choose a value appropriate
+// for your specific needs and network environment.
+func SetResolvingTimeout(timeout time.Duration) {
+	dns.ResolvingTimeout = timeout
```

Comment:
Ideally we'd have an end-to-end style tests that sets this and confirms it works as expected.

I think that would include overriding `internal.NewNetResolver` to produce a resolver that does nothing but block on the context, and then just doing `grpc.Dial("dns:///doesntmatter.com:123")` and waiting for the `State()` of the channel to change to `TransientFailure`.  It probably wouldn't be that hard to make - would you mind adding a test like that in this directory?